datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/build_traps_db.py

```diff
@@ -0,0 +1,294 @@
+)
+@click.option(
+    '--mib-sources',
+    '-s',
+    help='Url or a path to a directory containing the dependencies for [mib_files...].'
+    'Traps defined in these files are ignored.',
+)
+@click.option(
+    '--compiled-mibs-sources',
+    '-c',
+    help='Path to a directory where to store the intermediate compiled MIBs in JSON format. Defaults to a temp dir',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    help='Path to a directory where to store the created traps database file per MIB.'
+    'Recommended option, do not use with --output-file',
+)
+@click.option(
+    '--output-file',
+    help='Path to a file to store a compacted version of the traps database file. Do not use with --output-dir',
+)
+@click.option('--use-json', help='Use json instead of yaml for the output file(s).', is_flag=True)
+@click.argument(
+    'mib-files',
+    nargs=-1,
+)
+def build_traps_db(mib_sources, compiled_mibs_sources, output_dir, output_file, use_json, mib_files):
+    """Builds yaml formatted documents containing various information about traps. These file scan be used by
+    the Datadog Agent to enrich trap data.
+    This command is intended for "Network Devices Monitoring" users who need to enrich traps that are not automatically
+    supported by Datadog.
+
+    The expected workflow is as such:\n
+    1- Identify a type of device that is sending traps that Datadog does not already recognize.\n
+    2- Fetch all the MIBs that we do not support.\n
+    3- Run `ddev meta snmp build-traps-db -o ./output_dir/ /path/to/my/mibs/*`\n
+    """
+
+    # Defaulting to github.com/DataDog/mibs.snmplabs.com/
+    mib_sources = [mib_sources] if mib_sources else [MIB_SOURCE_URL]
+
+    if output_file:
+        allowed_extensions = ['.json'] if use_json else ['.yml', '.yml']
+        if not any(output_file.endswith(x) for x in allowed_extensions):
+            echo_warning(
+                "Output file {} does not end with an allowed extension '{}'".format(
+                    output_file, ", ".join(allowed_extensions)
+                )
+            )
+            output_file = output_file + allowed_extensions[0]
+            echo_warning("Using {} instead.".format(output_file))
+
+    with ExitStack() as stack:
+
+        if not compiled_mibs_sources:
+            compiled_mibs_sources = stack.enter_context(TempDir('ddev_mibs'))
+
+        compiled_mibs_sources = os.path.abspath(compiled_mibs_sources)
+        if not os.path.isdir(compiled_mibs_sources):
+            abort("Output directory {} does not exist".format(compiled_mibs_sources))
+        echo_info("Writing intermediate compiled MIBs to {}".format(compiled_mibs_sources))
+
+        mibs_sources_dir = os.path.join(compiled_mibs_sources, 'mibs_sources')
+        if not os.path.isdir(mibs_sources_dir):
+            os.mkdir(mibs_sources_dir)
+
+        if output_dir and output_file:
+            abort("Do not set both --output-dir and --output-file at the same time.")
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Instead of aborting you could override one over the other, goal to avoid fails as much as possible

Reply:
It's not clear how we should do that, and why one should have precedence over the other. So maybe our best option here to fail early?

Reply:
Could become a boolean option maybe, like the output format. Maybe by default dump to a single file, that can be override by `--output-type:[compact|per_mib]`

Reply:
Yeah I considered that but I thought it to be misleading:
1. We can have `--output-type` and `--output` but `output` refers either to a directory or a file. I feel it weird to have a given option (`--output`) to have a different use case based on another parameters.
2. We keep `--output-dir` and `--output-file` while also adding `--output-type` but I felt like the flag was not bringing anything.

What options did you have in mind? Maybe another I didn't think ok?